libminifi/include/controllers/NetworkPrioritizerService.h

```diff
@@ -121,6 +126,7 @@ class NetworkPrioritizerService : public core::controller::ControllerService, pu
   bool verify_interfaces_;
 
  private:
+  std::shared_ptr<utils::timeutils::Clock> clock_;
```

Comment:
Why shared_ptr, what are the other owners?

Reply:
in the tests the clock is being advanced externally

Reply:
The service and the unit test.  If only the service owned it, the unit test would need to have a naked pointer (or reference) to the clock, and rely on the order of the operations to make sure it remains valid -- I have considered this, but did not like it.  Since the `clock_` field is only created once, and never copied, there isn't much of an overhead.

Reply:
info(not ment to argue either for or against) : if we want to reference a member of a `shared_ptr` pointee, and not accidentally mess up, we can share the lifetime of the pointee while pointing to the member, so the "container" is kept alive even if we are only using its member